### PR TITLE
[docs] [explore] Restructuring explore section part-1

### LINKED ISSUE #10923: [Docs] Restructure certain sections under Explore
### Description

- Rename Distributed SQL features to SQL features.
- Since SQL features should comprise of SQL only features under it , move indexes after transactions.

Here's just how the ordering will possibly look like in **Explore** . @rkarthik007 , please validate.
> Explore
- SQL features
   1. Schemas and Tables
   2. Data Types
   3. Data Manipulation
   4. Queries and Joins
   5. Expressions and Operators
   6. Cursors
   7. Constraints
   8. Stored Procedures
   9. Triggers
   10. Table Partitioning
   11. Tablespaces
   12. Views
   13. Savepoints
   14. Collations
   15. Extensions

- Fault tolerance **( TODO: rename to Continuous Availability and add more content)**
- Horizontal scalability
- Transactions
- **Indexes** 
- **CDC (when we have it )**
- Cluster management 
- JSON support
- Multi-region deployments 
- Security
- Observability

### PR DESCRIPTION
**Pull request by aishwarya24**
Addresses part of  [10923](https://github.com/yugabyte/yugabyte-db/issues/10923)

This PR addresses the following:

- Renamed `Distributed SQL features` to `SQL features`
- SQL features also is re-ordered based on the ticket(10923) with sub-sections `Advanced features` and `Going beyond SQL`.
- Created `Indexes and constraints` section after `Transactions` with  Indexes, Constraints, and Generalized Inverted Indexes under it.(This part needs to be revisited later to match the necessary requirements for YSQL and YCQL tabs).
- Fixed ordering of files under `Observability` (Prometheus Integration first, followed by Grafana dashboard)
- Re-ordered Cluster management after Query tuning.
- Follower reads is moved to `Going beyond SQL` as per the requirements.
- Possible minor other changes. 

Note
 **Tablegroups** when documented, needs to go under `Going beyond SQL`.

Question

-  As per the requirement, `Data distribution with HASH` needs to be under `Going beyond SQL`. Currently the content for this page is under Horizontal scalability as [Sharding data](https://docs.yugabyte.com/latest/explore/linear-scalability/sharding-data/). Need clarity on handling this .
At the moment , `Follower reads` and `Tablespaces` are moved to the section.

@netlify /latest/explore/ysql-language-features/

### CODE REVIEW

**Comment by polarweasel** [comment]
Comment from Karthik: "Yes - link from the table to the Horizontal scalability section" (So, don't move sharding, just link to it from the overview table)

**Comment by aishwarya24** [comment]
> Comment from Karthik: "Yes - link from the table to the Horizontal scalability section" (So, don't move sharding, just link to it from the overview table)

Thanks for clarifying @polarweasel and @rkarthik007 . The table under **Going beyond SQL** links to it already: https://deploy-preview-11122--infallible-bardeen-164bc9.netlify.app/latest/explore/ysql-language-features/ 
From what I understand , this should be sufficient.

File: docs/content/latest/explore/indexes-constraints/indexes-1.md [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-headcontent: Indexes
+description: Using Indexes
 image: /images/section_icons/secure/create-roles.png
 menu:
   latest:
     identifier: indexes-1
-    parent: explore-ysql-language-features
-    weight: 250
-isTocNested: true
-showAsideToc: true
```
Comment (polarweasel) [inline docs/content/latest/explore/indexes-constraints/indexes-1.md:13]: Are you sure you don't want the on-page TOC? I think you should keep it...

File: docs/content/latest/explore/ysql-language-features/advanced-features/partitions.md [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
 linkTitle: Table Partitioning
 description: Table Partitioning in YSQL
-headcontent: Table Partitioning in YSQL
 image: /images/section_icons/secure/create-roles.png
 menu:
   latest:
-    identifier: explore-ysql-language-features-partitioins
-    parent: explore-ysql-language-features
-    weight: 315
+    identifier: advanced-features-partitioins
```
Comment (polarweasel) [inline docs/content/latest/explore/ysql-language-features/advanced-features/partitions.md:8]: Search for any other occurrences of the typo!
```suggestion
    identifier: advanced-features-partitions
```

File: docs/content/latest/explore/ysql-language-features/going-beyond-sql/follower-reads-ycql.md [hunk 3]
```diff
@@ -8,10 +8,11 @@ aliases:
   - /latest/explore/tunable-reads/
   - /latest/explore/follower-reads/
   - /latest/explore/high-performance/tunable-reads/
+  - /latest/explore/multi-region-deployments/follower-reads-ycql/
 menu:
   latest:
-    identifier: explore-multi-region-deployments-follower-reads-ycql
-    parent: explore-multi-region-deployments
+    identifier: going-beyond-sql-follower-reads-ycql
```
Comment (polarweasel) [inline docs/content/latest/explore/ysql-language-features/going-beyond-sql/follower-reads-ycql.md:14]: Might be good to shorten the identifier to `follower-reads-ycql` (and do the same for ysql). If nothing else, it'll be easier to move it some other time :)

File: docs/content/latest/explore/ysql-language-features/going-beyond-sql/tablespaces.md [hunk 4]
```diff
@@ -6,9 +6,11 @@ headcontent: Tablespaces in YSQL
 image: /images/section_icons/secure/create-roles.png
 menu:
   latest:
-    identifier: explore-ysql-language-features-tablespaces
-    parent: explore-ysql-language-features
+    identifier: going-beyond-sql-tablespaces
```
Comment (polarweasel) [inline docs/content/latest/explore/ysql-language-features/going-beyond-sql/tablespaces.md:9]: You're changing the identifier and parent in this file, but not in `tablespaces-platform.md` or `tablespaces-yugabyted.md` — is that on purpose?

**Review by polarweasel — CHANGES_REQUESTED** [review]

File: docs/content/latest/explore/ysql-language-features/_index.md [hunk 5]
```diff
@@ -26,47 +26,49 @@ The following diagram demonstrates how YugabyteDB reuses the PostgreSQL query la
...
 | :----------------------------------------------------------- | :----------------------------------------------------------- |
 | <span style="font-size:16px">[Data distribution with `HASH`](../linear-scalability/sharding-data/)</span> | Enables the use of `HASH` sort order, in addition to `ASC` and `DESC` for indexes |
-| <span style="font-size:16px">[`TABLESPACES` for geographic placement](tablespaces)</span> | Enables pinning of data in tables and table partitions to different geographic locations |
-| <span style="font-size:16px">`TABLEGROUPS` for colocating tables</span> | Enables colocation of multiple smaller tables into one tablet for better performance |
+| <span style="font-size:16px">[`Tablespaces`](going-beyond-sql/tablespaces)</span> | Enables pinning of data in tables and table partitions to different geographic locations |
+| <span style="font-size:16px">[`Follower Reads`](going-beyond-sql/follower-reads-ysql/)</span> | Enables more read IOPS with low latencies in YugabyteDB clusters |
+| <span style="font-size:16px">`Tablegroups`</span> | Enables colocation of multiple smaller tables into one tablet for better performance |
 
 <!--
```
[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/latest/explore/ysql-language-features/_index.md:73]: Thank you ! I was unsure about it.

File: docs/content/latest/explore/ysql-language-features/advanced-features/partitions.md (hunk 2 above)
Comment (aishwarya24) [reply docs/content/latest/explore/ysql-language-features/advanced-features/partitions.md:8]: Thank you! Yes , found in its stable version as well. Will make the change there too.

File: docs/content/latest/explore/ysql-language-features/going-beyond-sql/follower-reads-ycql.md (hunk 3 above)
Comment (aishwarya24) [reply docs/content/latest/explore/ysql-language-features/going-beyond-sql/follower-reads-ycql.md:14]: Understood. Will shorten it. Thanks.

File: docs/content/latest/explore/ysql-language-features/going-beyond-sql/tablespaces.md (hunk 4 above)
Comment (aishwarya24) [reply docs/content/latest/explore/ysql-language-features/going-beyond-sql/tablespaces.md:9]: So those two files are included as part of **Cluster creation** tab and their URL's looks like: /latest/explore/ysql-language-features/going-beyond-sql/tablespaces/#yugabyted
/latest/explore/ysql-language-features/going-beyond-sql/tablespaces/#platform

File: docs/content/latest/explore/indexes-constraints/indexes-1.md (hunk 1 above)
Comment (aishwarya24) [reply docs/content/latest/explore/indexes-constraints/indexes-1.md:13]: Missed to have the TOC set . Will include it. Thanks.

**Review by polarweasel — APPROVED** [review]
LGTM. And thanks for catching that identifier typo in stable, too!

**Pull request merged**